### PR TITLE
mongodb doc loader init

### PR DESCRIPTION
**Pull request by saginawj**
- **Description:** A Document Loader for MongoDB
  - **Issue:** n/a
  - **Dependencies:** Motor, the async driver for MongoDB
  - **Tag maintainer:** n/a
  - **Twitter handle:** pigpenblue

Note that an initial mongodb document loader was created 4 months ago, but the [PR ](https://github.com/langchain-ai/langchain/pull/4285)was never pulled in.  @leo-gan had commented on that PR, but given it is extremely far behind the master branch and a ton has changed in Langchain since then (including repo name and structure), I rewrote the branch and issued a new PR with the expectation that the old one can be closed.

Please reference that old PR for comments/context, but it can be closed in favor of this one.  Thanks!

### CODE REVIEW

**Comment by saginawj** [comment]
@eyurtsev would it be possible for you to take a look at this?  I'm installing 'motor' (the async engine for mongodb) as an optional install, then installing locally with 'poetry install -E all' since it's optional, yet the GitHub Actions don't seem to install dependencies in this way so the checks are failing.  I've tried to copy how other new packages are installed, and I read the CONTRIBUTING.MD and followed those instructions, yet still getting these issues.  thanks!

File: libs/langchain/langchain/document_loaders/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+class MongodbLoader(BaseLoader):
+    """Load MongoDB documents."""
+
+    def __init__(
+        self,
+        connection_string: str,
+        db_name: str,
+        collection_name: str,
+        filter_criteria: Optional[Dict] = None,
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/mongodb.py:19]: nit:
```suggestion
        *, 
        filter_criteria: Optional[Dict] = None,
```

File: libs/langchain/langchain/document_loaders/mongodb.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    ):
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/mongodb.py:20]: nit: Useful to document arguments

**Comment by eyurtsev** [comment]
@saginawj comandeering to resolve merge conflicts

File: libs/langchain/langchain/document_loaders/mongodb.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        self.collection_name = collection_name
+        self.filter_criteria = filter_criteria or {}
+
+        self.db = self.client.get_database(db_name)
+        self.collection = self.db.get_collection(collection_name)
+
+    def load(self) -> List[Document]:
+        """Load data into Document objects.
+
+        Attention:
```
Comment (eyurtsev) [inline libs/langchain/langchain/document_loaders/mongodb.py:48]: @saginawj i updated the description here. If you're interested, feel free to follow up with a PR to fix this implementation.

Steps:

1) Detect if there's a running loop
```
In [1]: import asyncio
In [2]: asyncio.get_running_loop()
```
2) 
-- If it doesn't exist then use asyncio.run with self.aload()
-- if it exists, strart a threadpool with 1 thread and have the thread run asyncio.run with self.aload()

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**